flink-core/src/main/java/org/apache/flink/configuration/BlobServerOptions.java

```diff
@@ -33,7 +33,7 @@
             key("blob.storage.directory")
                     .noDefaultValue()
```

Comment:
I'm wondering whether we could introduce a new "data type" for the file based options. Something along the lines of...
```suggestion
                    .fileType(FileType.LOCAL)
```

that would allow us to distinguish between local / flink filesystems and adjust the docs generator to work with that.

WDYT?

Reply:
What would that safe us & how would you enforce that all filepath options would use this filetype?

Reply:
Sorry, I've missed this one.

I'm not sure how to enforce it. We could have it return `URI` type, which could be a motivation for using it to a certain degree (the same logic kind of applies to why not have `int` values stored as `string`).

Anyway, it was just an idea, no need to block the PR on this.

Reply:
> have it return URI type

hmm, I'd argue that even options that only support local directories should support paths like `file:///tmp`. (Which currently some do, some don't, depending on who in the end ends up using it).